driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java

```diff
@@ -535,6 +536,11 @@ private void initClient(final BsonDocument entity, final String id,
                                         "Unsupported authMechanismProperties for authMechanism: " + value);
                             }
 
+                            // override the org.mongodb.test.uri connection string
```

Comment:
I don't understand why this is necessary.  Can't we always arrange to pass in the connection string as the system property?  Or are there secrets that we don't want to have logged?

Reply:
For OIDC, we need two different credentials - the OIDC credential for the test itself, and a credential that is able to perform admin operations, like dropping databases, that are required by our test infrastructure. Usually, the credential we use for the the test itself is also able to perform admin operations (and this must have been the case for OIDC as well, until a recent change).

Our prose tests already pull the credential for each test itself out of env variables, as per the spec, so only the unified tests needed to override this `org.mongodb.test.uri` to not use the admin credential.

Reply:
Yuk, but ok.